### ISSUE TITLE
Build failing with 4.0.0 : updateClassesForInheritance failed

### ISSUE THREAD

**Issue by eboudrant** [airbnb/epoxy#1054]
Since we updated to 4.0.0 we are seeing this error intermittently.

```
00:35:50 > Task :module:kaptDebugKotlin FAILED
00:35:50 e: error: EpoxyProcessorException: updateClassesForInheritance failed
00:35:50   com.airbnb.epoxy.processor.EpoxyProcessorException: updateClassesForInheritance failed
00:35:50   	at com.airbnb.epoxy.processor.Logger.logError(Logger.kt:63)
00:35:50   	at com.airbnb.epoxy.processor.Asyncable$DefaultImpls.awaitAndLog(Asyncable.kt:136)
00:35:50   	at com.airbnb.epoxy.processor.Asyncable$awaitAndLog$1.invokeSuspend(Asyncable.kt)
00:35:50   	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
00:35:50   	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
00:35:50   	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
00:35:50   	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
00:35:50   	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
00:35:50   	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
00:35:50   Caused by: java.util.ConcurrentModificationException
00:35:50   	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:911)
00:35:50   	at java.util.ArrayList$Itr.next(ArrayList.java:861)
00:35:50   	at com.airbnb.epoxy.processor.GeneratedModelInfo.removeMethodIfDuplicatedBySetter(GeneratedModelInfo.kt:116)
00:35:50   	at com.airbnb.epoxy.processor.GeneratedModelInfo.addAttributes(GeneratedModelInfo.kt:95)
00:35:50   	at com.airbnb.epoxy.processor.EpoxyProcessor$updateClassesForInheritance$2.invokeSuspend(EpoxyProcessor.kt:214)
00:35:50   	at com.airbnb.epoxy.processor.EpoxyProcessor$updateClassesForInheritance$2.invoke(EpoxyProcessor.kt)
00:35:50   	at com.airbnb.epoxy.processor.Asyncable$forEach$4$invokeSuspend$$inlined$map$lambda$1.invokeSuspend(Asyncable.kt:100)
00:35:50   	... 6 more
```

Let me know if you have an idea, I can send more info.

`Gradle 6.4.1,Android Gradle plugin 3.6.4, Kotlin 1.3.50`

In the meantime we disabled parallel processing.

**Comment by elihart** [comment]
Thanks for reporting! I should be able to figure this out from the stacktrace, and will hopefully have a fix soon

**Comment by elihart** [comment]
should be fixed in 4.1.0 https://github.com/airbnb/epoxy/pull/1055
